addons/delivery/models/delivery_carrier.py

```diff
@@ -186,6 +186,10 @@ def _get_delivery_type(self):
         self.ensure_one()
         return self.delivery_type
 
+    def _apply_margins(self, price):
+        self.ensure_one()
+        return float(price) * (1.0 + self.margin) + self.fixed_margin if self.delivery_type != 'fixed' else price
```

Comment:
Why do you need to cast price to float?
Also, nitpicking but should be <100 chars

Reply:
> Why do you need to cast price to float?

not sure to be honest I just left it in case there's a use case I'm not seeing (some carrier returning the price as string)

Reply:
From what I read, most delivery_providers cast the price in `*_rate_shipment`, and `_get_tax_included_unit_price` (just above the call to `_apply_margins`) returns a float. So I really doubt it is still useful. But I'd like the @odoo/rd-logistics team to confirm this.

Reply:
Anyway, you solved the issue with your update, so leave it iike that. Better not risk to break something in stable.